### ISSUE TITLE
Handle fontstack ascender/descender metadata

### ISSUE THREAD

**Issue by jseppi** [mapbox/glyph-pbf-composite#11]
Related to https://github.com/mapbox/node-fontnik/pull/160

When/if https://github.com/mapbox/node-fontnik/pull/160 is shipped, the new `ascender` and `descender` metadata fields will need to be handled here as well (unless the functionality of this repo is moved into `node-fontnik` -- https://github.com/mapbox/node-fontnik/pull/163).

We'll also need to figure out the desired behavior of combining ascender/descender metadata from the multiple fonts being composited. In discussing with @tristen, we think the compositing of that metadata should be: if all fonts in the composite have the same ascender/descender values, then use those, but if any are different, then use 0 (falling back to the current fixed value)

In summary:

- [ ] Add `ascender` and `descender` to `fontstack` in `glyphs.proto`, as in https://github.com/mapbox/node-fontnik/pull/160/files#diff-f5607f797a3301f210b1a4e9f7db4ffdR30-R31
- [ ] Implement `ascender`/`descender` compositing in `.composite()`

Ref:
- https://github.com/mapbox/mapbox-gl-native/pull/15676
- https://github.com/mapbox/mapbox-gl-js/pull/8781

**Comment by springmeyer** [comment]
> if all fonts in the composite have the same ascender/descender values, then use those, but if any are different, then use 0 (falling back to the current fixed value)

Have you tested this yet? Do most fonts use the same ascender/descender values such that the customized values will usually be respected? Or is it going to be very rare that we will not just zero them out?

**Comment by jseppi** [comment]
@springmeyer When fonts from the same family are composited, then I think the customized ascender/descender values are likely to be the same. For example, in this test case with two different "Lato" variations being composited: https://github.com/mapbox/glyph-pbf-composite/pull/12/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R132-R144

**Comment by jseppi** [comment]
@springmeyer Does this sound ok to you?

**Comment by springmeyer** [comment]
@jseppi Thanks for the explanation. It seems like that is fine assuming users have control over their entire fontstack and understand the implications of each font they choose in the fontstack.
